### PR TITLE
Create windows packaging

### PR DESCRIPTION
**Pull request by adriendelsalle**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x]  ~~Reset the build number to `0` (if the version changed)~~
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.
* [x] Fix `nlohmann_json` feedstock for broken dependency

Description
--
Related to [Windows packaging issue](https://github.com/conda-forge/json_schema_validator-feedstock/issues/1) where Windows packages are empty.

Proposition
--
Add a `bld.bat` file to handle Windows packaging

Discussion
--
@wolfv 
`nlohmann_json` feedstock is currently broken and need to be fixed before packaging this project (v3.8.0 required and not available on conda-forge).
I was not able to build locally and check the recipe due to this other issue.

### CODE REVIEW

**Comment by wolfv** [comment]
nlohmann json is updated now :)

**Comment by adriendelsalle** [comment]
@conda-forge-admin, please rerender

**Comment by adriendelsalle** [comment]
Thanks @wolfv it should be fixed now!
I also removed test executable from build to have minimal tarball

**Comment by wolfv** [comment]
Would be cool to add some tests to check that a dll is created and header files are placed in the right locations on Windows.

File: recipe/build.sh [hunk 1]
```diff
@@ -1,4 +1,10 @@
 #! /bin/bash
 
-cmake -DCMAKE_INSTALL_PREFIX=$PREFIX -DBUILD_SHARED_LIBS=y . -Bbuilddir
+cmake . \
+-DCMAKE_INSTALL_PREFIX=$PREFIX \
+-DBUILD_SHARED_LIBS=y \
```
Comment (wolfv) [inline recipe/build.sh]: maybe instead of `y` also use ON or TRUE?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -26,8 +26,9 @@ requirements:
 
 test:
   commands:
-    - test -f ${PREFIX}/include/nlohmann/json-schema.hpp  # [unix]
+    - test -f ${PREFIX}/include/nlohmann/json-schema.hpp
     - test -f ${PREFIX}/lib/libnlohmann_json_schema_validator.so  # [unix]
+    - test -f ${PREFIX}/lib/libnlohmann_json_schema_validator.dll  # [win]
```
Comment (wolfv) [inline recipe/meta.yaml]: might also be good to check %PREFIX%\bin\nlohmann_json_schema_validator.bin` (I think).

Also ${PREFIX} isn't gonna work on win

File: recipe/build.sh (hunk 1 above)
Comment (adriendelsalle) [reply recipe/build.sh]: I just copy/pasted from linux existing build script but if you prefer I can also change that

Comment (wolfv) [reply recipe/build.sh]: yep would be nice!

**Comment by wolfv** [comment]
can you also activate osx?

**Comment by wolfv** [comment]
Note the missing `lib`-prefix.

```
-- Installing: D:/bld/json_schema_validator_1621241783930/_h_env/Library/lib/nlohmann_json_schema_validator.lib
-- Installing: D:/bld/json_schema_validator_1621241783930/_h_env/Library/bin/nlohmann_json_schema_validator.dll
-- Installing: D:/bld/json_schema_validator_1621241783930/_h_env/Library/include/nlohmann/json-schema.hpp
```

**Pull request merged**